core/src/main/java/org/apache/custos/core/mapper/user/UserProfileMapper.java

```diff
@@ -181,6 +183,22 @@ public static org.apache.custos.core.user.profile.api.UserProfile createUserProf
             builder.setType(UserTypes.valueOf(profileEntity.getType()));
         }
 
+        if (profileEntity.getUserGroupMemberships() != null && !profileEntity.getUserGroupMemberships().isEmpty()) {
+            List<String> clientRoles = new ArrayList<>();
+            List<String> realmRoles = new ArrayList<>();
+            for (UserGroupMembership userGroupMembership: profileEntity.getUserGroupMemberships()) {
+                for (GroupRole gr: userGroupMembership.getGroup().getGroupRole()) {
+                    if (gr.getType().equals("client")) {
+                        clientRoles.add(gr.getValue());
+                    } else if (gr.getType().equals("realm")) {
```

Comment:
do the `.equals(..)` switching the constant and variable and make the `client` and `realm` be constants in the class

Reply:
Replaced with `gr.getType().equals(Constants.ROLE_TYPE_CLIENT)` and the `gr.getType().equals(Constants.ROLE_TYPE_REALM)`